src/txn/txn.c

```diff
@@ -702,6 +709,13 @@ __txn_commit_timestamps_assert(WT_SESSION_IMPL *session)
 	if (!F_ISSET(txn, WT_TXN_TS_COMMIT_KEYS | WT_TXN_TS_DURABLE_KEYS))
 		return (0);
 
+	cursors = NULL;
+	if (txn->mod_count != 0) {
+		WT_RET(__wt_calloc(
+		    session, sizeof(WT_CURSOR *), txn->mod_count, cursor_buf));
+		cursors = *cursor_buf;
```

Comment:
A more typical usage (used in the files I noted above) is to allocate into the local variable `&cursors` (maybe use `cursor_list` everywhere?) and then set the return variable after that is successful. The word `buf` doesn't help me though. Maybe something like `cursor_listp`?

Reply:
3009132

I've fixed the naming of these cursor variables to be more consistent.

As for the suggestion about only assigning to the return variable on success, I think that in this instance we should stray from the conventional usage. The reason is that the assertion function actually populates the update list with pointers to uncommitted data. With the suggested approach, if some assertion fails, we'll close the cursors and free that data making the update list unusable to the caller (walking will invoke UB).

Right now the difference is immaterial because if this assertion function returns non-zero, the caller won't be doing anything with the update lists. But it seems more error prone to me and I can see someone getting hit by this in the future.